airflow-core/src/airflow/ui/src/queries/useLogs.tsx

```diff
@@ -217,8 +220,23 @@ export const useLogs = (
     },
   );
 
+  const truncatedData = useMemo(() => {
+    if (!data?.content || typeof limit !== "number" || limit <= 0) {
+      return data;
+    }
+
+    const streamingContent = parseStreamingLogContent(data);
+    const truncatedContent =
+      streamingContent.length > limit ? streamingContent.slice(-limit) : streamingContent;
+
+    return {
+      ...data,
+      content: truncatedContent,
+    };
+  }, [data, limit]);
+
   const parsedData = parseLogs({
-    data: parseStreamingLogContent(data),
+    data: parseStreamingLogContent(typeof limit === "number" && limit > 0 ? truncatedData : data),
```

Comment:
`truncatedData` already holds that logic.

```suggestion
    data: parseStreamingLogContent(truncatedData),
```